### PR TITLE
Dispatcher csrf token

### PR DESCRIPTION
**Pull request by stojg**

### CODE REVIEW

**Comment by stojg** [comment]
please don’t merge just yet.. but  yarp or narp? I will need to fix all the other components in base projects

File: code/model/DNEnvironment.php [hunk 1]
```diff
@@ -654,10 +654,12 @@ protected function getCommitData($sha) {
...
 
 	/**
+	 * @param string $action
+	 *
 	 * @return string
 	 */
-	public function Link() {
-		return $this->Project()->Link() . "/environment/" . $this->Name;
+	public function Link($action = '') {
+		return \Controller::join_links($this->Project()->Link(), "environment", $this->Name, $action);
```
Comment (stojg) [inline code/model/DNEnvironment.php]: This is not strictly related to the PR, but I thought I would clean it up while I was around the neighbourhood

**Comment by halkyon** [comment]
Seems good to me @stojg
